source/__tests__/groupController.test.ts

```diff
@@ -0,0 +1,194 @@
+import { groups, mock } from "../models/__mocks__";
+import { Request, Response, NextFunction } from "express";
+import "regenerator-runtime/runtime";
+import GroupController from "../controllers/GroupController";
+
+describe("Group Controller", () => {
+  it("Get All Groups", async () => {
```

Comment:
try this code.

  it("Get All Groups", async () => {
    // const request = mock.request<typeof Request>();
    const request = mock.request();
    const response = mock.response();
    const next = mock.next();

    const groupService = {
      getGroups: jest.fn().mockResolvedValue({rows: groups}),
      getGroupById: jest.fn(),
      createGroup: jest.fn(),
      updateGroup: jest.fn(),
      deleteGroup: jest.fn(),
      addUsersToGroup: jest.fn(),
    };

    await new GroupController(groupService).getGroups({} as Request, response as unknown as Response, next);

    expect(response.json).toHaveBeenCalledTimes(1);
    expect(response.json).toHaveBeenCalledWith(groups);
  });

Reply:
Updated